### PR TITLE
DEV-3929-hard-conditions: add subject, test and solution for the exercise

### PR DESCRIPTION
**Pull request by MSilva95**

### CODE REVIEW

File: subjects/devops/hard-conditions/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+The `test` command, is a shell builtin that is used to evaluate expressions in a shell script. It has various options for performing different types of tests, such as checking the type of a file, comparing the values of two variables, or testing the status of a command.
+
+There are two syntaxes for using the test command.
+
+```console
+$ test EXPRESSION
+$ [ EXPRESSION ]
+```
+
+In a shell script, $1 is a special variable that refers to the first argument passed to the script. Arguments are values that are passed to the script when it is run, and they can be used to modify the behavior of the script or provide input to it.
```
Comment (mikysett) [inline subjects/devops/hard-conditions/README.md:32]: $1 -> `$1`

File: subjects/devops/hard-conditions/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Here is an example of a simple shell script that prints the value of $1:
```
Comment (mikysett) [inline subjects/devops/hard-conditions/README.md:34]: $1 -> `$1`

File: subjects/devops/hard-conditions/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#!/bin/bash
+
+# Print the value of $1
+echo "The first argument is: $1"
+```
+
+To run this script and pass an argument to it, you can use the following command:
+
+```console
+./script.sh arg1
```
Comment (mikysett) [inline subjects/devops/hard-conditions/README.md:46]: Here we can make the hint more explicit by doing something like:
```console
$ ./script.sh hello
The first argument is: hello
$
```
Notice I've changed `arg1` to something more random... I think students can be confused by the first argument being called `arg1`, they could think this is a special keyword to activate an argument or stuff like that.

File: sh/tests/hard-conditions/test-hard-conditions.sh [hunk 4]
```diff
@@ -0,0 +1 @@
+echo "hello world"
```
Comment (mikysett) [inline sh/tests/hard-conditions/test-hard-conditions.sh:1]: this file should be removed from the PR

File: sh/tests/hard-conditions_test.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+# Unofficial Bash Strict Mode
+set -euo pipefail
+IFS='
+'
```
Comment (mikysett) [inline sh/tests/hard-conditions_test.sh:6]: You can create the required file for tests like so:
```sh
echo "echo hello word" >test-hard-conditions.sh
```

**Review by mikysett — CHANGES_REQUESTED** [review]
Some little cleanups, but everything works already.

File: subjects/devops/hard-conditions/README.md (hunk 1 above)
Comment (MSilva95) [reply subjects/devops/hard-conditions/README.md:32]: done

File: subjects/devops/hard-conditions/README.md (hunk 2 above)
Comment (MSilva95) [reply subjects/devops/hard-conditions/README.md:34]: done

File: subjects/devops/hard-conditions/README.md (hunk 3 above)
Comment (MSilva95) [reply subjects/devops/hard-conditions/README.md:46]: done

File: sh/tests/hard-conditions/test-hard-conditions.sh (hunk 4 above)
Comment (MSilva95) [reply sh/tests/hard-conditions/test-hard-conditions.sh]: removed

File: sh/tests/hard-conditions_test.sh (hunk 5 above)
Comment (MSilva95) [reply sh/tests/hard-conditions_test.sh:6]: done

**Review by mikysett — APPROVED** [review]
perfect

**Pull request merged**